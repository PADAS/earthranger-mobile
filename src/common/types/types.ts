import { EventStatus, PatrolResult } from '../enums/enums';

export type Position = [number, number];

export type RootStackParamList = {
  SplashScreen: undefined,
  LoginView: undefined,
  TrackLocationMapView: {
    flyTo?: Position;
  }
  MainTabBar: undefined,
  ReportsView: undefined,
  ReportIssueView: undefined,
  PinAuthenticationView: undefined,
  SyncLoaderView: undefined,
  ReportCategoriesView: {
    coordinates?: Position,
  },
  ReportTypesView: {
    title: string,
    categoryId: string,
    coordinates?: Position,
    isPatrolInfoEventType?: boolean,
  },
  ReportForm: {
    reportId?: number,
    title: string,
    typeId: string,
    coordinates?: Position
    schema: string,
    geometryType: string,
    isEditMode?: boolean,
    createdAt?: string,
    isDefaultPatrolInfoEnabled?: boolean,
    geoJson?: GeoJSON.FeatureCollection,
    areaMeters?: number,
    perimeterMeters?: number,
    polygonPoints?: string,
    viewBox?: string,
    isPatrolInfoEvent?: boolean,
  },
  ReportEditLocationView: { coordinates: Position },
  RecordReportAreaView: undefined,
  AboutView: undefined,
  ReportNoteView: { id: number, text: string, coordinates: Position, channelId: string },
  MenuSettingsView: undefined,
  SettingsView: undefined,
  StatusView: undefined,
  ReportRepeatableFieldListView: {
    schema: any,
    title: string,
    rendererId: string,
    data?: RepeatableFieldListData[],
  },
  RepeatableFormView: {
    schema: any,
    title: string,
    listId: string,
  },
  EventsListFilterView: undefined,
  ReportsPendingSync: undefined,
  TrackedBySubjectsView: {
    subjects?: any,
  },
  BasemapView: undefined,
  CoordinateUnitsView: undefined,
  PatrolsTypeView: {
    coordinates: Position,
  },
  StartPatrolView: {
    patrolTitle: string,
    patrolType: string,
    patrolTypeId: string,
    coordinates: Position,
  },
  PhotoQualityView: undefined,
  ResetDatabaseCacheView: undefined,
  SubjectsListView: {
    parentSubjectGroupName?: string,
    isParentView?: boolean,
    parentId?: string,
  }
};

export type BottomSheetParamList = {
  Basemap: undefined,
  PatrolDetailsView: { patrolId: number },
  PatrolTrack: undefined,
  TrackingStatusView: undefined,
  SubjectDetailsView: undefined,
};

export interface EdgeInsets {
  top: number;
  right: number;
  bottom: number;
  left: number;
}

export type RepeatableFieldListData = {
  id: number;
  title: string;
  data: any;
  hidden: boolean,
};

export type LastLocation = {
  timestamp: string;
  displayDate: string;
  latitude: number;
  longitude: number;
  accuracy: number;
};

export interface IconProps {
  width?: string;
  height?: string;
  color?: string;
  viewbox?: string;
  isDisabled?: boolean;
}

export type AttachmentPhotoType = 'photo';
export type AttachmentNoteType = 'note';
export type AttachmentUpdateNoteType = 'updateNote';
export type AttachmentAddNewNoteType = 'addNewNote';
export type AttachmentDeleteNoteType = 'deleteNote';

export type ReportFormEvent = {
  id: number;
  remoteId: string,
  accountId: string,
  eventTypeId: number;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string;
  event_values: string;
  isDraft: number;
  createdAt: number,
  updatedAt: number,
  patrol_segment_id: number | undefined,
};

export type EventData = {
  eventTypeId: string;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string;
  event_values: string;
  isDraft: number;
  createdAt: string,
  updatedAt: string,
  patrol_segment_id: number | undefined,
};

export type ReportFormFullEvent = {
  account_id: number;
  profile_id: number;
  event_type_id: string;
  title: string;
  latitude: number;
  longitude: number;
  geometry: string,
  event_values: string;
  is_draft: number;
  value: string;
  updated_at: string;
  created_at: string;
  id: string;
  geometry_type: string;
  segment_remote_id: string | undefined;
  patrol_segment_id: string | undefined;
};

export type PersistedEventType = {
  id?: number;
  remote_id: string;
  account_id: number;
  profile_id: string;
  value: string;
  display: string;
  schema: string;
  category_id: string;
  geometry_type: string;
  default_priority: string;
  icon: string;
  icon_svg: string;
};

export type PersistedPatrolType = {
  id: number;
  remote_id: string;
  account_id: number;
  value: string;
  display: string;
  icon: string;
  icon_svg: string;
  default_priority: number;
  is_active: number;
  is_selected: boolean;
};

export type PersistedPatrol = {
  id: number;
  remote_id: string;
  account_id: number;
  profile_id: number;
  title: string;
  priority: string;
  state: string;
  created_at: string;
  updated_at: string;
};

export type PersistedPatrolSegment = {
  id: number
  remote_id: string;
  patrol_id: number;
  patrol_type_id: number;
  start_latitude: string;
  start_longitude: string;
  stop_latitude: string;
  stop_longitude: string;
  start_time: string;
  end_time: string;
};

export type PersistedPatrolTypeSegment = {
  id: number;
  remote_id: string;
  patrol_id: number;
  patrol_type_id: number;
  start_latitude: string;
  start_longitude: string;
  stop_latitude: string;
  stop_longitude: string;
  start_time: string;
  end_time: string;
  patrol_type: string;
};

export type DraftReports = {
  id: number;
  title: string;
  type_id: number,
  updated_at: string;
  icon_svg: string;
  default_priority: number;
  schema: string,
  geometry_type: string,
  hidden?: boolean;
  is_draft?: boolean;
};

export type EventUploadDetails = {
  reportId: number;
  reportTitle: string;
  isReportUploaded: boolean;
  uploaded: {
    notes: number;
    images: number;
  };
  pending: {
    notes: number;
    images: number;
  };
  state: string;
};

export type ReportFormEventAttachments = {
  id: number,
  remote_id: string,
  profile_id: number,
  account_id: number,
  event_id: string;
  type: string;
  path: string;
  thumbnail_path: string,
  note_text: string;
  uploaded: number,
  attachment_id: string;
};

export type PersistedPatrolDetail = {
  remote_id: string,
  title: string,
  serial_number: string,
  start_latitude: number,
  start_longitude: number,
  start_time: number,
  icon_svg: number,
  created_at: number,
};

export type PatrolDetail = {
  remoteId: string,
  title: string,
  serialNumber: string,
  startLatitude: number,
  startLongitude: number,
  startTime: number,
  iconSVG: string
  startTimeFormatted: string,
  createdAt: number,
};

export type PersistedUserProfile = {
  id: number;
  remote_id: string;
  account_id: string;
  username: string;
  content_type: string;
  subject_id: string;
  pin: string;
  permissions: string,
};

export type UserInfo = {
  id: number,
  type: string,
  permissions: string,
};

export type PersistedUserSubject = {
  id: number;
  remote_id: string;
  account_id: string;
  name: string;
  content_type: string;
  isSelected: boolean;
};

export type Patrol = {
  id: number,
  typeId: string,
  type: string,
  display: string,
  coordinates: Position | null,
  status: PatrolResult,
  timestamp: string,
};

// Polygon Reducer

export type PolygonState = {
  past: Position[],
  current: Position[],
};

export type PolygonAction = {
  type: 'UNDO' | 'REDO' | 'ADD',
  value?: Position,
};

type TGeometry = 'polygon' | 'point';

export const Geometry: Record<TGeometry, string> = {
  polygon: 'polygon',
  point: 'point',
};

export type Size = {
  width: number,
  height: number,
};

export interface EventListItem {
  bgColor: string,
  defaultPriority: number,
  error: string,
  fgColor: string,
  hidden: boolean,
  icon: string,
  id: number,
  isDraft: boolean,
  labelText: string,
  remoteId: string,
  status: EventStatus,
  statusIcon: string,
  text: string,
  title: string,
  errorMsg?: string,
  hasErrorOnEvent: boolean,
}

export type SubjectGroupData = {
  count: number,
  isVisible: boolean,
  title: string,
  type: 'group',
  subjects: number[],
  id: string,
};

export type Subject = {
  id: number,
  name: string,
  type: 'subject',
  isVisible: boolean,
  icon: string,
  lastPosition: Position,
  lastPositionUpdate: string,
  isHidden: boolean,
};

export type SubjectGroupSyncItem = {
  name: string,
  remoteId: string,
  parentId?: string,
  parentLocalId?: string,
};

export type SubjectSyncItem = {
  iconSVG: string,
  isActive: string,
  lastPosition: string,
  lastPositionDate: string,
  name: string,
  remoteId: string,
  tracksAvailable: string,
  updatedAt: string,
};
