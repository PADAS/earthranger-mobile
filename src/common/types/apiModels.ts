export interface FileRequest {
  uri: string,
  type: string,
  name: string,
}

export enum ApiStatus {
  Unknown = 1,
  Succeeded = 200,
  BadRequest = 400,
  Unauthorized = 401,
  Forbidden = 403,
  NotFound = 404,
  ServerError = 500,
}
