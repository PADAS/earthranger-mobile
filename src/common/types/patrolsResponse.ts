import { Subject } from './usersResponse';

export interface Status {
  code: number;
  message: string;
}

export interface PatrolType {
  default_priority: number;
  display: string;
  icon_id: string;
  id: string;
  is_active: boolean;
  value: string;
  ordernum: number;
}

export interface PatrolTypesResponse {
  data: PatrolType[];
  status: Status;
}

export interface TrackPatrolByResponse {
  data: {
    properties: {
      leader: {
        enum: Subject [],
      }
    }
  }
  status: Status,
}
