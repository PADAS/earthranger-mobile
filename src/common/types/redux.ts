export interface InitialState {
  session: {
    accessToken: string,
    refreshToken: string,
    error: null,
    sessionError: null,
  },
  reports: {
    noteId: number,
    textNote: string,
  }
}

export type ActionType = 'REFRESH_TOKEN'
| 'REFRESH_TOKEN_ERROR'
| 'USER_LOGIN'
| 'USER_LOGIN_ERROR'
| 'GET_SESSION'
| 'GET_SESSION_ERROR'
| 'CREATE_NOTE'
| 'CLEAR_NOTE_DATA'
| 'NUMBER_OF_REPORT_DRAFTS'
| 'CLEAR_SESSION';

export interface ReducerAction {
  type: ActionType,
  payload: any;
}
