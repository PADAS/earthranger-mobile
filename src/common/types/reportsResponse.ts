export interface ReportCategoriesResponse {
  data: Category[];
  status: Status;
}

export interface Category {
  id: string;
  value: string;
  display: string;
  ordernum: number;
  flag?: string;
  account_id: string | undefined;
  profile_id: number[];
}

export interface ReportTypesResponse {
  data: Type[];
  status: Status;
}

export interface ReportTypeSchemaResponse {
  data: string;
  status: Status;
}

export interface Type {
  id: string;
  icon: string;
  value: string;
  display: string;
  ordernum: number;
  is_collection: boolean;
  readonly?: boolean;
  category: Category;
  icon_id: string;
  is_active: boolean;
  default_priority: number;
  geometry_type: string;
  schema: string;
  url: string;
  icon_svg: string;
}

export interface Status {
  code: number;
  message: string;
}

export interface EventCategory {
  id: number;
  remote_id: string;
  value: string;
  display: string;
}

export interface EventType {
  id: number;
  display: string;
  icon_svg: string;
  default_priority: string;
}

export interface Note {
  id: number,
  text: string,
}

export interface WithChildren {
  children: any;
}

export interface UploadedReport {
  id: number;
}
