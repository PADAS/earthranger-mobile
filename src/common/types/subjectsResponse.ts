export interface SubjectGroupResponse {
  data: SubjectGroup[],
  status: Status,
}

export interface SubjectGroup {
  name: string,
  id: string,
  subgroups: SubjectGroup[],
  subjects: Subject[],
}

export interface Subject {
  content_type: string,
  id: string,
  name: string,
  subject_type: string,
  subject_subtype: string,
  common_name: string | null,
  additional: SubjectAdditional,
  created_at: string,
  updated_at: string,
  is_active: boolean,
  user: null,
  region: string,
  country: string,
  sex: string,
  tracks_available: boolean,
  image_url: string,
  last_position_status: SubjectLastPositionStatus,
  last_position_date: string,
  last_position: SubjectLastPosition,
  device_status_properties: SubjectDeviceStatusProperties[],
  url: string,
}

interface SubjectAdditional {
  rgb: string,
  sex: string,
  region: string,
  country: string,
  external_id: string,
  tm_animal_id: string,
  external_name: string,
}

interface SubjectLastPositionStatus {
  last_voice_call_start_at: string | null,
  radio_state_at: string | null,
  radio_state: string,
}

interface SubjectLastPosition {
  type: string,
  geometry: {
    type: string,
    coordinates: [number, number]
  },
  properties: {
    title: string,
    subject_type: string,
    subject_subtype: string,
    id: string,
    stroke: string,
    stroke_opacity: number,
    stroke_width: number,
    image: string,
    last_voice_call_start_at: string | null,
    location_requested_at: string | null,
    radio_state_at: string,
    radio_state: string,
    coordinateProperties: {
      time: string,
    },
    DateTime: string,
  }
}

interface SubjectDeviceStatusProperties {
  label: string,
  units: string,
  value: number,
}

interface Status {
  code: number;
  message: string;
}
