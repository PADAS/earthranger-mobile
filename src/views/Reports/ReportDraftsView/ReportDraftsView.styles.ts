// External Dependencies
import { StyleSheet } from 'react-native';

// Internal Dependencies
import { COLORS_LIGHT } from '../../../common/constants/colors';
import { IS_ANDROID } from '../../../common/constants/constants';
import { fontSize, fontWeight, lineHeight } from '../../../common/constants/fonts';

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: COLORS_LIGHT.white,
  },
  flatList: {
    marginBottom: IS_ANDROID ? 45 : 30,
    backgroundColor: COLORS_LIGHT.white,
  },
  appbar: {
    backgroundColor: COLORS_LIGHT.white,
  },
  titleAppBar: {
    fontSize: fontSize.large,
    lineHeight: lineHeight.large,
    fontWeight: fontWeight.normal,
    color: COLORS_LIGHT.G0_black,
  },
  dateHeaderText: {
    marginLeft: 'auto',
    marginRight: 16,
    marginTop: IS_ANDROID ? 16 : 0,
    fontSize: fontSize.medium,
    lineHeight: lineHeight.medium,
    fontWeight: fontWeight.normal,
    color: COLORS_LIGHT.G0_black,
  },
  divider: {
    height: 1,
    marginTop: 13,
    backgroundColor: COLORS_LIGHT.G5_LightGreyLines,
  },
});

export default styles;
