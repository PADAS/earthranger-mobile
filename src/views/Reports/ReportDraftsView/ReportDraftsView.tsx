/* eslint-disable arrow-body-style */
// External Dependencies
import React, {
  useCallback,
  useEffect,
  useRef,
  useState,
} from 'react';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useTranslation } from 'react-i18next';
import { FlatList, View, Text } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Incubator } from 'react-native-ui-lib';
import dayjs from 'dayjs';

// Internal Dependencies
import { ReportDraftsListCell } from './components/ReportDraftsListCell/ReportDraftsListCell';
import { useRetrieveReportPendingSync } from '../../../common/data/reports/useRetrieveReportPendingSync';
import log from '../../../common/utils/logUtils';
import { DraftReports, Position, RootStackParamList } from '../../../common/types/types';
import { useRemoveReportDraft } from '../../../common/data/reports/useRemoveReportDraft';
import { COLORS_LIGHT } from '../../../common/constants/colors';
import { TrashIcon as TrashIconSVG } from '../../../common/icons/TrashIcon';
import { UndoIcon as UndoIconSVG } from '../../../common/icons/UndoIcon';
import { customBackButton, osBackIcon } from '../../../common/components/header/header';
import { logEvent } from '../../../analytics/wrapper/analyticsWrapper';
import { createOpenReportDraftEvent, removeReportDraftEvent, undoDeleteDraftEvent } from '../../../analytics/reports/reportsAnalytics';
import AnalyticsEvent, { analyticsEventToHashMap } from '../../../analytics/model/analyticsEvent';

// Styles
import styles from './ReportDraftsView.styles';

// Icons
const TrashIcon = () => <TrashIconSVG />;
const UndoIcon = () => <View style={{ marginRight: 8 }}><UndoIconSVG width="16" /></View>;

// Interfaces + Types
interface ReportDraftsViewProps {
  navigation: NativeStackNavigationProp<RootStackParamList, 'ReportDraftsView'>;
}

const ReportDraftsView = ({ navigation }: ReportDraftsViewProps) => {
  // Hooks
  const { t } = useTranslation();
  const { retrieveDraftReports } = useRetrieveReportPendingSync();
  const { removeReportDraft } = useRemoveReportDraft();

  // Component's State
  const [draftReports, setDraftReports] = useState<DraftReports[]>([]);
  const [displayToast, setDisplayToast] = useState(false);
  const [removedReportDraftId, setRemovedReportDraftId] = useState<number | null>();
  const shouldRemoveReportDraft = useRef(false);

  useEffect(() => {
    const initDraftReports = async () => {
      try {
        const draftReportsList = await retrieveDraftReports();
        setDraftReports(draftReportsList);
      } catch (error) {
        log.debug(`[ReportDraftsView] :: Could not retrieve draft reports - Error ${error}`);
      }
    };

    initDraftReports();
  }, []);

  useEffect(() => {
    navigation.setOptions({
      // @ts-ignore
      headerLeft: () => customBackButton(osBackIcon, onBackButton, false),
      gestureEnabled: false,
    });
  }, [displayToast]);

  // Handlers
  const onRemoveReportHandler = async (id: number) => {
    trackAnalyticsEvent(removeReportDraftEvent());
    draftReports.forEach((draftReport) => {
      if (draftReport.id === id) {
        // eslint-disable-next-line no-param-reassign
        draftReport.hidden = true;
        setRemovedReportDraftId(id);
        shouldRemoveReportDraft.current = true;
      }
    });

    setDisplayToast(true);
    setDraftReports(draftReports);
  };

  const onRemoveReportUndo = () => {
    trackAnalyticsEvent(undoDeleteDraftEvent());
    draftReports.forEach((draftReport) => {
      if (draftReport.id === removedReportDraftId) {
        // eslint-disable-next-line no-param-reassign
        draftReport.hidden = false;
        setRemovedReportDraftId(null);
        shouldRemoveReportDraft.current = false;
      }
    });
  };

  const onBackButton = () => {
    if (!displayToast) {
      navigation.goBack();
    }
    return null;
  };

  // Utilities
  const commitReportRemoval = async () => {
    if (!shouldRemoveReportDraft.current || !removedReportDraftId) {
      setDisplayToast(false);
      return;
    }

    try {
      const rowsAffected = await removeReportDraft(removedReportDraftId.toString());

      if (rowsAffected === 1) {
        const filteredDraftReports = draftReports.filter((draftReport) => {
          return draftReport.id !== removedReportDraftId;
        });

        setDisplayToast(false);
        setDraftReports(filteredDraftReports);
      }
    } catch (error) {
      log.debug(`[ReportDraftsView] :: onRemoveReportHandler - Error: ${error}`);
    }
  };

  const navigateToReportForm = useCallback((
    reportId: number,
    title: string,
    typeId: string,
    schema: string,
    geometryType: string,
    isEditMode: true,
  ) => {
    navigation.navigate('ReportForm', {
      reportId,
      title,
      typeId,
      coordinates: [0, 0] as Position,
      schema,
      geometryType,
      isEditMode,
    });
  }, []);

  const trackAnalyticsEvent = (event: AnalyticsEvent) => {
    logEvent(event.eventName, analyticsEventToHashMap(event));
  };

  return (
    <SafeAreaView style={styles.safeArea}>
      <View>
        <Text style={styles.dateHeaderText}>{t('reportDrafts.dateUpdated')}</Text>
        <View style={styles.divider} />
        <FlatList
          style={styles.flatList}
          data={draftReports}
          keyExtractor={(item) => item.id.toString()}
          renderItem={({ item }) => {
            if (!item?.hidden) {
              return (
                <ReportDraftsListCell
                  iconImage={item.icon_svg}
                  title={item.title}
                  date={dayjs(item.updated_at).format('D MMM \'YY HH:mm')}
                  priority={item.default_priority.toString()}
                  onPress={() => {
                    trackAnalyticsEvent(createOpenReportDraftEvent());
                    navigateToReportForm(
                      item.id,
                      item.title,
                      item.type_id.toString(),
                      item.schema,
                      item.geometry_type,
                      true,
                    );
                  }}
                  onRemove={() => onRemoveReportHandler(item.id)}
                />
              );
            }
            return null;
          }}
        />
      </View>

      {/* Report Delete Confirmation Toast */}
      <Incubator.Toast
        action={{
          color: COLORS_LIGHT.white,
          iconSource: UndoIcon,
          label: `${t('common.undo')}`,
          onPress: () => onRemoveReportUndo(),
          style: {
            backgroundColor: 'transparent',
            marginRight: 16,
          },
        }}
        autoDismiss={2000}
        backgroundColor={COLORS_LIGHT.G0_black}
        icon={TrashIcon}
        message={t('reportDrafts.toastRemoveConfirmationText')}
        messageStyle={{ color: COLORS_LIGHT.white }}
        onDismiss={() => commitReportRemoval()}
        position="bottom"
        style={{ borderRadius: 0 }}
        visible={displayToast}
      />
      {/* End Report Delete Confirmation Toast */}
    </SafeAreaView>
  );
};

export { ReportDraftsView };
