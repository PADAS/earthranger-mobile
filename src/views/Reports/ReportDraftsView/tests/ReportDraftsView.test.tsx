// External Dependencies
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

// Internal Dependencies
import { ReportDraftsView } from '../ReportDraftsView';
import { RootStackParamList } from '../../../../common/types/types';

// Types + Interfaces
// eslint-disable-next-line max-len
type StackNavigationPropAlias = NativeStackNavigationProp<RootStackParamList, 'ReportTypesView'>;

// Test Specific Mocks
jest.mock('../../../../common/data/reports/useRetrieveReportPendingSync', () => ({
  useRetrieveReportPendingSync: () => ({
    retrieveDraftReports: () => Promise.resolve([{
      id: '1',
      title: 'Reptile',
      updated_at: '1667529163154',
      icon_svg: '',
      default_priority: 0,
    }]),
  }),
}));

describe('ReportDraftsView', () => {
  let navigation: Pick<StackNavigationPropAlias, 'setOptions' >;

  beforeEach(() => {
    navigation = {
      setOptions: jest.fn(),
    };
  });

  it('matches snapshot', async () => {
    const { toJSON, getByText } = render(
      // @ts-ignore
      <ReportDraftsView navigation={navigation} />,
    );

    await waitFor(() => {
      expect(getByText('Reptile')).toBeTruthy();
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
