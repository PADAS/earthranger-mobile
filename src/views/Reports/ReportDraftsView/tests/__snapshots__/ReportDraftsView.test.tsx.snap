// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDraftsView matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 14,
          "marginLeft": "auto",
          "marginRight": 16,
          "marginTop": 0,
        }
      }
    >
      Date updated
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#DDDDDD",
          "height": 1,
          "marginTop": 13,
        }
      }
    />
    <RCTScrollView
      data={
        Array [
          Object {
            "default_priority": 0,
            "icon_svg": "",
            "id": "1",
            "title": "Reptile",
            "updated_at": "1667529163154",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "marginBottom": 30,
        }
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={1}
              handlerType="PanGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onLayout={[Function]}
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#D0031B",
                    "bottom": 0,
                    "flexDirection": "row-reverse",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": -10000,
                      },
                    ],
                  }
                }
              >
                <View
                  collapsable={false}
                  row={true}
                  style={
                    Object {
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <RNGestureHandlerButton
                    activeOpacity={0.105}
                    collapsable={false}
                    delayLongPress={600}
                    handlerTag={2}
                    handlerType="NativeViewGestureHandler"
                    onActiveStateChange={[Function]}
                    onGestureEvent={[Function]}
                    onGestureHandlerEvent={[Function]}
                    onGestureHandlerStateChange={[Function]}
                    onHandlerStateChange={[Function]}
                    onPress={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#D0031B",
                        "justifyContent": "center",
                        "minWidth": undefined,
                        "paddingHorizontal": 12,
                        "width": undefined,
                      }
                    }
                    underlayColor="black"
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "backgroundColor": "black",
                          "borderBottomLeftRadius": undefined,
                          "borderBottomRightRadius": undefined,
                          "borderRadius": undefined,
                          "borderTopLeftRadius": undefined,
                          "borderTopRightRadius": undefined,
                          "bottom": 0,
                          "left": 0,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                    <Image
                      collapsable={false}
                      source={
                        Object {
                          "testUri": "../../../assets/icons/ic-trash.png",
                        }
                      }
                      style={
                        Object {
                          "height": 24,
                          "marginHorizontal": 10,
                          "opacity": 0.2,
                          "tintColor": "#FFFFFF",
                          "transform": Array [
                            Object {
                              "scale": 0.2,
                            },
                          ],
                          "width": 24,
                        }
                      }
                    />
                    <Text
                      accessibilityElementsHidden={true}
                      accessible={false}
                      collapsable={false}
                      importantForAccessibility="no-hide-descendants"
                      style={
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 12,
                          "opacity": 0.2,
                          "transform": Array [
                            Object {
                              "scale": 0.2,
                            },
                          ],
                        }
                      }
                    >
                      Delete
                    </Text>
                  </RNGestureHandlerButton>
                </View>
                <View
                  onLayout={[Function]}
                />
              </View>
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={3}
                handlerType="TapGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "translateX": -0,
                      },
                    ],
                  }
                }
              >
                <View
                  accessibilityActions={
                    Array [
                      Object {
                        "label": "Delete",
                        "name": "Delete",
                      },
                    ]
                  }
                  accessible={true}
                  onAccessibilityAction={[Function]}
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Object {},
                      Object {},
                      Object {},
                      undefined,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "flex": 1,
                          "flexDirection": "row",
                          "height": 58,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        }
                      }
                    >
                      <View>
                        <RNSVGSvgView
                          bbHeight="40"
                          bbWidth="40"
                          focusable={false}
                          height="40"
                          style={
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                                "borderWidth": 0,
                              },
                              Object {
                                "flex": 0,
                                "height": 40,
                                "width": 40,
                              },
                            ]
                          }
                          width="40"
                        >
                          <RNSVGGroup>
                            <RNSVGGroup>
                              <RNSVGGroup>
                                <RNSVGGroup
                                  fill={4294967295}
                                  opacity={0.8}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                >
                                  <RNSVGCircle
                                    cx="7.5"
                                    cy="7.5"
                                    r="6.75"
                                  />
                                  <RNSVGPath
                                    d="M7.5 1.5a6 6 0 1 1-6 6 6 6 0 0 1 6-6m0-1.5A7.5 7.5 0 1 0 15 7.5 7.508 7.508 0 0 0 7.5 0Z"
                                  />
                                </RNSVGGroup>
                                <RNSVGCircle
                                  cx={20}
                                  cy={20}
                                  fill={4284704362}
                                  propList={
                                    Array [
                                      "fill",
                                    ]
                                  }
                                  r={20}
                                />
                              </RNSVGGroup>
                            </RNSVGGroup>
                          </RNSVGGroup>
                        </RNSVGSvgView>
                      </View>
                      <Text
                        numberOfLines={2}
                        style={
                          Object {
                            "fontSize": 16,
                            "fontWeight": "500",
                            "lineHeight": 18,
                            "marginLeft": 12,
                            "textAlignVertical": "center",
                            "width": "54%",
                          }
                        }
                      >
                        Reptile
                      </Text>
                      <Text
                        style={
                          Object {
                            "fontSize": 12,
                            "fontWeight": "400",
                            "lineHeight": 14,
                            "marginLeft": "auto",
                          }
                        }
                      >
                        2 Jul '71 15:15
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#DDDDDD",
                          "height": 1,
                          "margingTop": 10,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
